### PR TITLE
Adding AppCompatCache

### PR DESCRIPTION
**Pull request by secabstraction**
Let me know if this needs more work, plz/thx ;)

### CODE REVIEW

**Comment by secabstraction** [comment]
Oh man, I didn't include any pester tests 👎 I'll write them, if that's the bar for entry.

File: CimSweep/ArtifactRetrieval/NetworkProfiles.ps1 [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                                2 { 'Domain' }
+                            }
+                        }
+                        
+                        'Managed' { $ObjectProperties['Managed'] = [bool]$ValueContent }
+                          
+                          default { $ObjectProperties[$ValueName] = $ValueContent }
+                    }
+                }
+                [PSCustomObject]$ObjectProperties
```
Comment (mattifestation) [inline CimSweep/ArtifactRetrieval/NetworkProfiles.ps1]: Hey Jesse. You have an OutputType specified of CimSweep.NetworkProfile but there's no PSTypeName applied to $ObjectProperties. Get that added and I'll merge in. Everything else looks good. Thanks!

-Matt

**Review by mattifestation — COMMENTED** [review]
Is the following output to be expected in any cases?

```
Path             : 00000009     000106a66d710000        000a000039d00000        8664    Microsoft.MSPaint 8wekyb3d8bbwe
LastModifiedTime : 1601-01-01T00:00:00.0000000Z
```

File: CimSweep/ArtifactRetrieval/NetworkProfiles.ps1 (hunk 1 above)
Comment (mattifestation) [reply CimSweep/ArtifactRetrieval/NetworkProfiles.ps1]: I'm on Win10 Enterprise RS2 for reference.

Comment (mattifestation) [reply CimSweep/ArtifactRetrieval/NetworkProfiles.ps1]: A good Pester test for Get-CSAppCompatCache would be to perform a sanity check on each returned path as well as the date.

**Comment by secabstraction** [comment]
So, as far as I can tell, that is the way appcompat stores entries for
trusted store apps. I'm not sure how to parse what looks like possibly
other data values stored in the string. Any ideas there?

I'll add the TypeName for the NetworkProfile, and write the tests ;)

File: CimSweep/ArtifactRetrieval/NetworkProfiles.ps1 (hunk 1 above)
Comment (mattifestation) [reply CimSweep/ArtifactRetrieval/NetworkProfiles.ps1]: Ah yeah I see that @EricZimmerman's [AppCompatCacheParser](https://github.com/EricZimmerman/AppCompatCacheParser) doesn't handle Store apps either. Without digging in, I recognize these fields:

* 8664 == X64 architecture
* Microsoft.MSPaint == AppX name
* 8wekyb3d8bbwe == PublisherId
* 000106a66d710000 == Version - 0x0001 (1), 0x06a6 (1702), 0x6d71 (28017), 0x0 (0) == 1.1702.28017.0
* 000a000039d00000 == Version (not sure what this corresponds to) - 10.0.14800.0

The first field might refer to a length. For example, the field will be 0xB when there's a ResourceId included (e.g. neutral).

So you could translate much of this into the PackageFullName: `Microsoft.MSPaint_2.1704.26037.0_x64__8wekyb3d8bbwe`

There are also some that include a ResourceId (e.g. neutral). Therefore, the format would look like: `<Name>_<Version>_<Architecture>_<ResourceId>_<PublisherId>`

You can check your work with:
```powershell
Get-AppxPackage -Name 'AppName'
```

Comment (EricZimmerman) [reply CimSweep/ArtifactRetrieval/NetworkProfiles.ps1]: Are the store apps kept in the same value? im happy to add support or the extra fields, any test data available or a process to add some to the value?

Comment (EricZimmerman) [reply CimSweep/ArtifactRetrieval/NetworkProfiles.ps1]: ok so i looked into this again and based on my unit tests my stuff is processing things out.

With that said, i can certainly clean things up and make the path mean a bit more vs what shows up now.

image

Comment (mattifestation) [reply CimSweep/ArtifactRetrieval/NetworkProfiles.ps1]: Thanks for the info @EricZimmerman. Personally, I'm fine leaving the entries as is until all of the fields for Store apps can be figured out.

**Comment by secabstraction** [comment]
I've done some more research here and the best I can resolve from the
fields present is the package family name. The rest of the fields don't
seem to be reliable enough to consistently resolve the full name.

**Pull request merged**